--- a/tools/cli/main.js	2025-10-16 23:50:50.027404730 +0200
+++ b/tools/cli/main.js	2025-10-16 23:50:50.027404730 +0200
@@ -515,6 +515,44 @@
   }
 
   // Now exec; we're not coming back.
+
+  // BEGIN HACK
+  // patch shebang:
+  var fs = require('fs');
+  var path = require("path")
+  var srcOld = fs.readFileSync(executable, 'utf8');
+  srcNew = srcOld.replace(/^#!\/bin\/bash/, '#!/bin/sh');
+  if (srcOld !== srcNew) {
+    fs.writeFileSync(executable, srcNew, 'utf8');
+  }
+  // patch elfs:
+  var dir = path.dirname(executable);
+  var interpreter = "@INTERPRETER@";
+  var rpath = "@RPATH@";
+  function spawnSync(/*...*/) {
+    var args = Array.prototype.slice.call(arguments);
+    var proc = require("child_process").spawn.apply(null, args);
+    return new Promise(function(resolve) {
+      proc.on('close', function (code) {
+        resolve();
+      });
+    });
+  }
+  function patchelf(/*...*/) {
+    var pathParts = Array.prototype.slice.call(arguments);
+    var p = path.join.apply(null, [dir, "dev_bundle"].concat(pathParts));
+    return spawnSync('@PATCHELF@', [
+      "--set-interpreter",
+      interpreter,
+      "--set-rpath",
+      rpath,
+      p
+    ], {stdio: 'inherit'});
+  }
+  await patchelf("bin", "node");
+  await patchelf("mongodb", "bin", "mongod");
+  await patchelf("mongodb", "bin", "mongos");
+  // END HACK
   require('kexec')(executable, newArgv);
   throw Error('exec failed?');
 };
